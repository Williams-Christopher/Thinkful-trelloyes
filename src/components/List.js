import React from 'react';

import './List.css';
import Card from './Card';

function List(props) {
    //console.log(`props: ${JSON.stringify(props)}`);

    // Create an array of Card components
    let listCards = props.cards.map(card =>
        <Card id={card.id} onDeleteCard={props.onDeleteCard} title={card.title} content={card.content} />
    );
    
    return (
        <section className='List'>
            <header className='List-header'>
                <h2>{props.header}</h2>
            </header>
            <div className='List-cards'>
                {listCards}
                <button className='list-add-button' type='submit' name='cardAdd' onClick={() => props.onAddCard(props.id)}>Add Random</button>
            </div>
        </section>
    );
}

export default List;