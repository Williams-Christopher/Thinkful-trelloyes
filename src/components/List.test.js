import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';

describe('List Component Tests', () => {
    const testProps = {
        header: 'Test List',
        cards: [{key: 12345, title: 'Test card', content: 'Test content'}],
    }

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<List header={testProps.header} cards={testProps.cards}/>, div);
        ReactDOM.unmountComponentAtNode(div);
        });

    it('Renders the UI as expected', () => {
        const tree = renderer.create(<List header={testProps.header} cards={testProps.cards} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});