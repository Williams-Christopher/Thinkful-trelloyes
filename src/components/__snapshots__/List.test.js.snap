// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component Tests Renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      Test List
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        Test card
      </h3>
      <p>
        Test content
      </p>
    </div>
  </div>
</section>
`;
